src/main/java/com/emc/pravega/perf/PravegaTransactionWriterWorker.java

```diff
@@ -34,30 +34,33 @@
     PravegaTransactionWriterWorker(int sensorId, int eventsPerSec,
                                    int secondsToRun, boolean isRandomKey,
                                    int messageSize, Instant start,
-                                   ClientFactory factory, PerfStats stats,
-                                   String streamName,
-                                   long totalEvents,
+                                   PerfStats stats, String streamName,
+                                   long totalEvents, ClientFactory factory,
                                    int transactionsPerCommit) {
 
         super(sensorId, eventsPerSec, secondsToRun, isRandomKey,
-                messageSize, start, factory, stats, streamName, totalEvents);
+                messageSize, start, stats, streamName, totalEvents, factory);
 
         this.transactionsPerCommit = transactionsPerCommit;
         eventCount = new AtomicInteger(0);
         transaction = new AtomicReference<Transaction<String>>(producer.beginTxn());
     }
 
     @Override
-    public CompletableFuture writeData(String key, String data) throws TxnFailedException, IllegalStateException {
+    public CompletableFuture writeData(String key, String data) throws IllegalStateException {
         final Transaction<String> curTrans = transaction.get();
 
-        curTrans.writeEvent(key, data);
-        if (eventCount.incrementAndGet() >= transactionsPerCommit) {
-            eventCount.set(0);
-            curTrans.commit();
-            if (!transaction.compareAndSet(curTrans, producer.beginTxn())) {
-                throw new IllegalStateException("WriteData called on the same PravegaTransactionWriterWorker from two threads in parallel.");
+        try {
+            curTrans.writeEvent(key, data);
+            if (eventCount.incrementAndGet() >= transactionsPerCommit) {
+                eventCount.set(0);
+                curTrans.commit();
+                if (!transaction.compareAndSet(curTrans, producer.beginTxn())) {
+                    throw new IllegalStateException("WriteData called on the same PravegaTransactionWriterWorker from two threads in parallel.");
+                }
             }
+        } catch (TxnFailedException e) {
+            throw new IllegalStateException("Write data failed with transaction failure.");
```

Comment:
I'm not sure this qualfies as an illegal state.

Reply:
yes, I do feel so.. but can you suggest which generic exception i can use? shall i use RuntimeException?

Reply:
RuntimeException is fine, just be sure to pass `e` as the second arg to the constructor so that the trace and cause is not lost.

Reply:
yes , its done

Reply:
done